nncf/common/pruning/mask_propagation.py

```diff
@@ -110,6 +111,10 @@ def symbolic_mask_propagation(self, prunable_layers_types: List[str],
                     assert len(input_masks) == 1
                     input_mask = input_masks[0] # type: SymbolicMask
 
+                    # find input_mask for node if multiple input masks existed
+                    if isinstance(input_mask, list):
+                        input_mask = find_input_mask_for_node(input_mask, node)
+
```

Comment:
Can we resolve this inside `get_input_masks`?

Reply:
I also thought that, but in some case we use `get_input_masks` to get all input masks(multiple input case), literally, but other cases like this, we tried to get a input mask (singe input case). So If we resolve this inside `get_input_masks`, it could not cover both of those cases.

Reply:
Can we do something like this?

```
def get_input_masks(node, graph) -> List[NNCFTensor]:
    retval = []
    input_masks = [input_node.data['output_mask'] for input_node in graph.get_previous_nodes(node)]
    for input_mask in input_masks:
        retval.append(input_mask[node.node_name] if isinstance(input_mask, dict) else input_mask)
    return retval
```

Len of retval equal to amount of input edges, so we can resolve input mask for each input edge

Reply:
Fix.